core/domain/suggestion_services.py

```diff
@@ -2217,55 +2251,125 @@ def update_translation_contribution_stats(
-        translation_contribution_stat.accepted_translations_count += 1 * int(not is_submission and is_suggestion_accepted)
-        translation_contribution_stat.accepted_translations_without_reviewer_edits_count += 1 + int(
-            not is_submission and suggestion.edited_by_reviewer)
+
+def update_translation_contribution_stats_at_review(
+    suggestion: suggestion_registry.BaseSuggestion,
+    user_id: str, target_id: str) -> None:
+    """Creates/updates TranslationContributionStatsModel model for
+    given translation submitter when a translation is reviewed.
+
+    Args:
+        suggestion: Suggestion. The suggestion domain object that is being
+            reviewed.
+        user_id: str. The ID of the corresponding submitter.
+        target_id: str. The ID of the target entity being suggested to.
+    """
+    topic_id = ''
+    exp_opportunity_dict = (
+        opportunity_services.get_exploration_opportunity_summaries_by_ids(
+            [target_id]))
+    exp_opportunity = exp_opportunity_dict[target_id]
+    # Ruling out the possibility of None for mypy type checking.
+    assert exp_opportunity is not None
+    topic_id = exp_opportunity.topic_id
+
+    content_plain_text = html_cleaner.strip_html_tags(
+        suggestion.change.translation_html)
+    content_word_count = len(content_plain_text.split())
+
+    translation_contribution_stat_model = (
+        suggestion_models.TranslationContributionStatsModel.get(
+            suggestion.change.language_code, user_id, topic_id
+        ))
+
+    if translation_contribution_stat_model is None:
+        suggestion_models.TranslationContributionStatsModel.create(
+            language_code=suggestion.change.language_code,
+            contributor_user_id=user_id,
+            topic_id=topic_id,
+            submitted_translations_count=1,
+            submitted_translation_word_count=content_word_count,
+            accepted_translations_count=0,
+            accepted_translations_without_reviewer_edits_count=0,
+            accepted_translation_word_count=0,
+            rejected_translations_count=0,
+            rejected_translation_word_count=0,
+            contribution_dates=[
+                datetime.datetime.strptime(
+                    str(
+                        suggestion.last_updated.utcnow().date().isoformat()),
+                        '%Y-%m-%d'
+                    ).date()
+            ]
+        )
+    else:
+        translation_contribution_stat = (
+            create_translation_contribution_stats_from_model(
+                translation_contribution_stat_model))
+
+        suggestion_is_accepted = (
+        suggestion.status == suggestion_models.STATUS_ACCEPTED)
+
+        suggestion_is_rejected = (
+            suggestion.status == suggestion_models.STATUS_REJECTED)
+        (
+            translation_contribution_stat
+            .accepted_translations_count
+        ) += 1 * int(suggestion_is_accepted)
```

Comment:
Re https://github.com/oppia/oppia/pull/15917#discussion_r948653058 -- I just meant, why `1 * int(X)` instead of just `int(X)`? Ditto a few lines below.

Reply:
Yeah just `int(X)` is sufficient and updated.